tfjs-backend-webgpu/src/backend_webgpu.ts

```diff
@@ -483,18 +483,134 @@ export class WebGPUBackend extends KernelBackend {
     }
   }
 
+  private makeUniformsDataView(data: DataView): GPUBindingResource {
+    const dimensionsBuffer = this.acquireBuffer(
+        data.byteLength, GPUBufferUsage.COPY_DST | GPUBufferUsage.UNIFORM);
+    this.queue.writeBuffer(dimensionsBuffer, 0, data);
+
+    return {offset: 0, size: data.byteLength, buffer: dimensionsBuffer};
+  }
+
+  private numberOrArrayToDataView(
+      arrays: Array<{type: string; data: number | number[]}>,
+      length: number): DataView {
+    const BYTES_PER_ELEMENT = 4;
+    const uniformDataView =
+        new DataView(new ArrayBuffer(length * BYTES_PER_ELEMENT));
+
+    let dataViewIndex = 0;
+    arrays.forEach(array => {
+      const arrayData = array.data;
+
+      if (array.type !== 'int32' && array.type !== 'float32') {
+        throw new Error(`${array.type} not supported!`);
+      }
+
+      if (Array.isArray(arrayData)) {
+        if (array.type === 'int32') {
+          arrayData.forEach(d => {
+            uniformDataView.setInt32(
+                dataViewIndex * BYTES_PER_ELEMENT, d, true);
+            dataViewIndex++;
+          });
+        } else {
+          arrayData.forEach(d => {
+            uniformDataView.setFloat32(
+                dataViewIndex * BYTES_PER_ELEMENT, d, true);
+            dataViewIndex++;
+          });
+        }
+      } else {
+        if (array.type === 'int32') {
+          uniformDataView.setInt32(
+              dataViewIndex * BYTES_PER_ELEMENT, arrayData, true);
+          dataViewIndex++;
+        } else {
+          uniformDataView.setFloat32(
+              dataViewIndex * BYTES_PER_ELEMENT, arrayData, true);
+          dataViewIndex++;
+        }
+      }
+    });
+
+    return uniformDataView;
+  }
+
+  private computePadding(uniformsWithType:
```

Comment:
Because we had already added padding in this function. Is it better to rename it as "addPaddingToFollowStd140"?

Reply:
Currently there is only one kind of padding is computed. Mayebe we can rename it when we have more.